### PR TITLE
Properly translate a bunch of strings

### PR DESCRIPTION
**Pull request by esoergel**
@sravfeyn @millerdev @czue

### CODE REVIEW

File: docs/translations.rst [hunk 1]
```diff
@@ -3,12 +3,33 @@ Internationalization
...
+* ``ugettext``: The function immediately returns the translation for the
+  currently selected language.
+* ``ugettext_lazy``: The function converts the string to a translation
+  "promise" object.  This is later coerced to a string when rendering a
+  template or otherwise forcing the promise.
+* ``ugettext_noop``: This function only marks a string as translation string,
+  it does not have any other effect; that is, it always returns the string
+  itself. This should be considered an advanced tool and generally avoided.
+  It could be useful if you need access to both the translated and untranslated
+  strings.
```
Comment (sravfeyn) [inline docs/translations.rst]: Thanks for these awesome notes!

File: docs/translations.rst [hunk 2]
```diff
@@ -3,12 +3,33 @@ Internationalization
...
-`Django Docs translations page <https://docs.djangoproject.com/en/dev/topics/i18n/translation/>`_.
+`Django Docs translations page <https://docs.djangoproject.com/en/dev/topics/i18n/translation/>`_
+or `this helpful blog post <http://blog.bessas.me/post/65775299341/using-gettext-in-django>`_.
 
 
 Tagging strings in views
 ------------------------
 
+**TL;DR**: ``ugettext`` should be used in code that will be run per-request.
+``ugettext_lazy`` should be used in top-level code.
```
Comment (millerdev) [inline docs/translations.rst]: The meaning of "top-level code" isn't very clear, although I can infer from the context that it is code that is run at module import time or generally outside of a request. Aside: do celery tasks have a locale?

File: docs/translations.rst [hunk 3]
```diff
@@ -19,10 +40,10 @@ The most common case is just wrapping text with ugettext.
         messages.success(request, _("Welcome!"))
 
 
-Sometimes it isn't always so simple, for example with report or settings
-section names. Two methods (`ugettext_noop` and `ugettext_lazy`) are available to mark
-a string for translation but not store the translated string value. These solve
-slightly different cases but are often interchangeable.
+When a string will be read by python as soon as the code is parsed, there is
```
Comment (millerdev) [inline docs/translations.rst]: Would it be clear to say "Typically when code is run as a result of a module being imported, ..."?

File: docs/translations.rst [hunk 4]
```diff
@@ -69,18 +106,18 @@ work properly.
...
-    class BaseUserInfoForm(forms.Form):
-        first_name = forms.CharField(label=ugettext_lazy('First Name'), max_length=50, required=False)
-        last_name = forms.CharField(label=ugettext_lazy('Last Name'), max_length=50, required=False)
+    >>> import json
+    >>> from django.utils.translation import ugettext_lazy
+    >>> json.dumps({"message": ugettext_lazy("Hello!")})
+    TypeError: <django.utils.functional.__proxy__ object at 0x7fb50766f3d0> is not JSON serializable
+    >>> from dimagi.utils.web import json_handler
+    >>> json.dumps({"message": ugettext_lazy("Hello!")}, default=json_handler)
+    '{"message": "Hello!"}'
```
Comment (millerdev) [inline docs/translations.rst]: Nice!

File: docs/translations.rst (hunk 2 above)
Comment (esoergel) [inline docs/translations.rst]: that's what I mean.  Not sure how best to phrase it, and would welcome edits if you're so inclined.

Celery tasks I think don't have a locale unless specified.  For instance, scheduled reports do something like
```python
with localize(user.language):
    generate_report()
```

File: docs/translations.rst (hunk 3 above)
Comment (esoergel) [inline docs/translations.rst]: :+1:

**Pull request merged**